packages/p2p/src/components/advertiser-page/advertiser-page-adverts/advertiser-page-row.tsx

```diff
@@ -0,0 +1,191 @@
+    counterparty_type: string;
+    country: string;
+    created_time: number;
+    description: string;
+    effective_rate: null | number;
+    effective_rate_display: string;
+    id: string;
+    is_active: number;
+    is_visible: number;
+    local_currency: string;
+    max_order_amount_limit: number;
+    max_order_amount_limit_display: string;
+    min_order_amount_limit: number;
+    min_order_amount_limit_display: string;
+    payment_method: string;
+    payment_method_names: string[];
+    price: number | null;
+    price_display: string;
+    rate: number;
+    rate_display: string;
+    rate_type: string;
+    type: string;
+};
+
+export type TAdvertiserPageRow = {
+    row: TAdvertiserPageDetails;
+};
+
+const AdvertiserPageRow = ({ row: advert }: TAdvertiserPageRow) => {
+    const { advertiser_page_store, buy_sell_store, floating_rate_store, general_store } = useStores();
+    const {
+        client: { currency },
+    } = useStore();
+    const {
+        effective_rate,
+        local_currency,
+        max_order_amount_limit_display,
+        min_order_amount_limit_display,
+        payment_method_names,
+        price_display,
+        rate_type,
+        rate,
+    } = advert;
+    const { showModal } = useModalManagerContext();
+
+    const is_buy_advert = advertiser_page_store.counterparty_type === buy_sell.BUY;
+    const is_my_advert = advertiser_page_store.advertiser_details_id === general_store.advertiser_id;
+
+    const { display_effective_rate } = generateEffectiveRate({
+        price: price_display,
+        rate_type,
+        rate,
+        local_currency,
+        exchange_rate: floating_rate_store.exchange_rate,
+        market_rate: effective_rate,
+    });
+
+    const showAdForm = () => {
+        buy_sell_store.setSelectedAdState(advert);
+        showModal({
+            key: 'BuySellModal',
+        });
+    };
+
+    const getButtonLabel = () => `${is_buy_advert ? localize('Buy') : localize('Sell')} ${currency}`;
+
+    const getButtonContent = () =>
+        is_my_advert ? (
+            <Table.Cell />
+        ) : (
+            <Table.Cell className='advertiser-page__adverts-button'>
+                <Button
+                    is_disabled={general_store.is_barred}
+                    onClick={showAdForm}
+                    primary
+                    small={isDesktop()}
+                    large={isMobile()}
+                >
+                    {getButtonLabel()}
+                </Button>
+            </Table.Cell>
+        );
+
+    return (
+        <React.Fragment>
+            <DesktopWrapper>
+                <Table.Row className='advertiser-page__adverts-table_row'>
+                    <Table.Cell>{`${min_order_amount_limit_display}-${max_order_amount_limit_display} ${currency}`}</Table.Cell>
+                    <Table.Cell className='advertiser-page__adverts-price'>
+                        <Text color='profit-success' line-height='m' size='xs' weight='bold'>
```

Comment:
non-blocking: remember to remove any of the default props such as the line-height

```suggestion
                        <Text color='profit-success' size='xs' weight='bold'>
```